presto-main/src/test/java/io/prestosql/type/TestArrayOperators.java

```diff
@@ -597,13 +597,13 @@ public void testArrayMin()
         assertFunction("ARRAY_MIN(ARRAY [NaN()])", DOUBLE, NaN);
         assertFunction("ARRAY_MIN(ARRAY [NULL, NULL, NULL])", UNKNOWN, null);
         assertFunction("ARRAY_MIN(ARRAY [NaN(), NaN(), NaN()])", DOUBLE, NaN);
-        assertFunction("ARRAY_MIN(ARRAY [NULL, 2, 3])", INTEGER, 2);
+        assertFunction("ARRAY_MIN(ARRAY [NULL, 2, 3])", INTEGER, null);
         assertFunction("ARRAY_MIN(ARRAY [NaN(), 2, 3])", DOUBLE, 2.0);
-        assertFunction("ARRAY_MIN(ARRAY [NULL, NaN(), 1])", DOUBLE, 1.0);
-        assertFunction("ARRAY_MIN(ARRAY [NaN(), NULL, 3.0])", DOUBLE, 3.0);
-        assertFunction("ARRAY_MIN(ARRAY [1.0E0, NULL, 3])", DOUBLE, 1.0E0);
+        assertFunction("ARRAY_MIN(ARRAY [NULL, NaN(), 1])", DOUBLE, null);
```

Comment:
Before c95e937517f773ba625cfde38aac7ed11e6f8493 the expected values here were different -- here and in some other places NaN was expected. See

https://github.com/prestosql/presto/blob/f04fa5ca8de1015629a94693740dfbc641e55784/presto-main/src/test/java/io/prestosql/type/TestArrayOperators.java#L608-L612

Reply:
You're right, though that seems like a different issue, and a bug. It's not clear why NaN would be returned over NULL, but non-NaN values would return NULL.

Reply:
According to the documentation, it is a bug:

> Like most other functions in Presto, they return null if any argument is null.

Reply:
I think the logic can be "return NaN when any of the compared values is NaN" and "NULL means unknown, so return unknown result when any of the compared values is unknown but, unless the result will be NaN regardless of the unknown value"

Reply:
Anyway, even if want to change the behavior, I'd strongly prefer to restore pre-344 behavior to clear "release blocker" and then fix whatever remains to be fixed on top of that.

Reply:
Actually, that quote is from greatest/least. The only documentation for array_min is

> Returns the minimum value of input array.